app/javascript/lib/error-message.js

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react'
+
+
+/** Hook for enabling the conditional display an error message
+  * in the event of an error
+  * the returned ErrorComponent can be placed in the component's markup, and will be
+  * displayed with the content from setErrorContent if setShowError is called with true
+  */
```

Comment:
Less runny, and more conventional form:

```suggestion
/** 
  * Hook for enabling the conditional display an error message.
  * In the event of an error, the returned ErrorComponent can be placed in the component's markup, 
  * and will be displayed with the content from setErrorContent if setShowError is called with true.
  */
```